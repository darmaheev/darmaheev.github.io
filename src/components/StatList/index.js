import React from "react";
import PropTypes from "prop-types";

import {STAT_LIST} from "../../constants/common";
import StatItem from "../StatItem";
import {
  Wrapper
} from "./styles";

const initialSumValue = 0;

const StatList = ({
  stats
}) => (
  <Wrapper>
    {Object.keys(STAT_LIST).map(stat => (<StatItem key={STAT_LIST[stat].value} label={STAT_LIST[stat].label}
      value={stats.find(pS => pS.stat.name === STAT_LIST[stat].value).base_stat}/>))}
    <StatItem label={"Total"} value={stats.reduce((acum, stat) => (acum + stat.base_stat), initialSumValue)}/>
  </Wrapper>
);


StatList.defaultProps = {
  stats: [
    {base_stat: 0, stat: {name: "hp"}},
    {base_stat: 0, stat: {name: "attack"}},
    {base_stat: 0, stat: {name: "special-attack"}},
    {base_stat: 0, stat: {name: "defense"}},
    {base_stat: 0, stat: {name: "special-defense"}}
  ]
};

StatList.propTypes = {
  stats: PropTypes.arrayOf(PropTypes.shape({
    base_stat: PropTypes.number,
    stat: PropTypes.shape({
      name: PropTypes.string
    })
  }))
};

export default StatList;