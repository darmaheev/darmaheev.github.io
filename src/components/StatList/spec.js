import React from "react";
import { MemoryRouter } from "react-router-dom";
import StatList from "./index";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders statList with default props", () => {
  const statList = shallow(
    <StatList/>
  );
  expect(statList).toMatchSnapshot();
});

it("renders statList with data", () => {
  const stats = [
    {base_stat: 10, stat:{name: "hp"}},
    {base_stat: 20, stat:{name: "attack"}},
    {base_stat: 30, stat:{name: "special-attack"}},
    {base_stat: 40, stat:{name: "defense"}},
    {base_stat: 50, stat:{name: "special-defense"}}
  ];

  const searchList = mount(
    <MemoryRouter>
      <StatList
        stats={stats}
      />
    </MemoryRouter>
  );
  expect(searchList.text()).toContain("10");
  expect(searchList.text()).toContain("20");
  expect(searchList.text()).toContain("30");
  expect(searchList.text()).toContain("40");
  expect(searchList.text()).toContain("50");
  expect(searchList.text()).toContain("150");
});
