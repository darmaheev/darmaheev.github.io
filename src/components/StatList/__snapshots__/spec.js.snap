// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders statList with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatList
    stats={
      Array [
        Object {
          "base_stat": 0,
          "stat": Object {
            "name": "hp",
          },
        },
        Object {
          "base_stat": 0,
          "stat": Object {
            "name": "attack",
          },
        },
        Object {
          "base_stat": 0,
          "stat": Object {
            "name": "special-attack",
          },
        },
        Object {
          "base_stat": 0,
          "stat": Object {
            "name": "defense",
          },
        },
        Object {
          "base_stat": 0,
          "stat": Object {
            "name": "special-defense",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        Array [
          <StatItem
            label="HP"
            value={0}
          />,
          <StatItem
            label="Atk"
            value={0}
          />,
          <StatItem
            label="Def"
            value={0}
          />,
          <StatItem
            label="SpA"
            value={0}
          />,
          <StatItem
            label="SpD"
            value={0}
          />,
        ],
        <StatItem
          label="Total"
          value={0}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "hp",
        "nodeType": "function",
        "props": Object {
          "label": "HP",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "attack",
        "nodeType": "function",
        "props": Object {
          "label": "Atk",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "defense",
        "nodeType": "function",
        "props": Object {
          "label": "Def",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "special-attack",
        "nodeType": "function",
        "props": Object {
          "label": "SpA",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "special-defense",
        "nodeType": "function",
        "props": Object {
          "label": "SpD",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Total",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex: 1;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  border-top: 1px solid #ebedef;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          Array [
            <StatItem
              label="HP"
              value={0}
            />,
            <StatItem
              label="Atk"
              value={0}
            />,
            <StatItem
              label="Def"
              value={0}
            />,
            <StatItem
              label="SpA"
              value={0}
            />,
            <StatItem
              label="SpD"
              value={0}
            />,
          ],
          <StatItem
            label="Total"
            value={0}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "hp",
          "nodeType": "function",
          "props": Object {
            "label": "HP",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "attack",
          "nodeType": "function",
          "props": Object {
            "label": "Atk",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "defense",
          "nodeType": "function",
          "props": Object {
            "label": "Def",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "special-attack",
          "nodeType": "function",
          "props": Object {
            "label": "SpA",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "special-defense",
          "nodeType": "function",
          "props": Object {
            "label": "SpD",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Total",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex: 1;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  border-top: 1px solid #ebedef;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
